go/serial/foreign_key.fbs

```diff
@@ -0,0 +1,56 @@
+// Copyright 2022 Dolthub, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+namespace serial;
+
+// https://dev.mysql.com/doc/refman/8.0/en/create-table-foreign-keys.html#foreign-key-referential-actions
+enum ForeignKeyReferentialAction : uint8 {
+  DefaultAction = 0,
+  Cascade = 1,
+  NoAction = 2,
+  Restrict = 3,
+  SetNull = 4,
+}
+
+table ForeignKeyCollection {
+  foreign_keys:[ForeignKey];
+}
+
+table ForeignKey {
+  // foreign key name
+  name:string;
+
+  // child table
+  child_table_name:string;
+  child_table_index:string;
+  child_table_columns:[uint64];
+
+  // parent table
+  parent_table_name:string;
+  parent_table_index:string;
+  parent_table_columns:[uint64];
+
+  // reference options
+  on_update:ForeignKeyReferentialAction;
+  on_delete:ForeignKeyReferentialAction;
+
+  // unresolved details
+  unresolved_child_columns:[string];
+  unresolved_parent_columns:[string];
```

Comment:
These used to represent unresolved columns, but now they're always set. Before the migration I am planning to fully remove the `child_table_columns:[uint64];` and `parent_table_columns:[uint64];` fields (in addition to the index fields too) and strictly rely on these fields instead. I'd just call these `string_` for now.